### ISSUE TITLE
Accessing missing properties should give an early sequelize error

### ISSUE THREAD

**Issue by manast** [sequelize/sequelize#6558]
This may be a fundamental design flaw in Sequelize and impossible to fix, but I think it is important to document it somewhere.

Basically the problem is when trying to refer to a table column that does not exist in the model. Sequelize does not seem to care to validate its existence before doing a query. This results in the database giving an exception. The problem with this approach is that I loose the ability to know where in my code this invalid query is being performed. I guess the underlying issue here is that since .sync is not really syncing the models with SQL, the developers of Sequelize decided to delegate the validation to the SQL server instead. I do not think this is the correct approach, it seems to be wiser to make the validation before making the query, independently if that column exists in SQL or not.

**Comment by mickhansen** [comment]
Are you expecting sequelize to know about columns in the database that you haven't defined in the code?
I'm curious where you would have gotten that expectation since docs clearly show you need to define your models.

**Comment by manast** [comment]
I updated the explanation above since I noticed I wrote it wrongly.

**Comment by mickhansen** [comment]
Makes more sense. While at times a requested feature Sequelize does not validate itself against a db to generate a diff, validations or anything else.

Building a diff between model and remote is on the todo list, but not highly prioritized.

**Comment by manast** [comment]
you don't need to make a diff to resolve this issue. Its just a matter of checking if the fields that you are using on the query, where clause for instance, are really part of the model, and if not give an exception.

**Comment by mickhansen** [comment]
No, with your updated description you are asking Sequelize to check that a where field is defined in the model but not in the DB.

Either way we support a lot of complicated where setups so enforcing that they must match a field would be error prone.

**Comment by manast** [comment]
No. I am not asking Sequelize to check this situation, I am just trying to describe what happens when you have a property defined in the model but not in the DB. Sequelize should not care if a field exists or not in the DB, just in the model, if it does not exist provide an error.

**Comment by mickhansen** [comment]
Wouldn't be impossible, but it hasn't been a vocal issue anywhere and i see a lot of possible errors with the possible magic where combinations we provide.

**Comment by mickhansen** [comment]
I'd be happy to keep this issue open as a feature request if you reword the title and provide an EDIT update to the description it'll be clear what feature is being asked in the future.

**Comment by manast** [comment]
ok.

**Comment by manast** [comment]
It is a feature request.
